internal/node/indexer/server.go

```diff
@@ -48,21 +49,30 @@ type Server struct {
 
 func (s *Server) Run(ctx context.Context) error {
 	var (
-		// TODO Develop a more effective solution to implement back pressure.
-		tasksChan = make(chan *engine.Tasks)
+		// TODO Develop a more effective solution to implement back pressure. - Resolving using utils.TaskBuffer
 		errorChan = make(chan error)
 	)
 
 	zap.L().Info("start node", zap.String("version", constant.BuildVersion()))
 
-	s.source.Start(ctx, tasksChan, errorChan)
+	s.source.Start(ctx, s.TaskBuffer, errorChan)
 
 	for {
 		select {
-		case tasks := <-tasksChan:
+		case err := <-errorChan:
+			if err != nil {
+				return fmt.Errorf("an error occurred in the source: %w", err)
+			}
+
+			return nil
```

Comment:
```suggestion
The return nil statement immediately after the if err != nil check is unnecessary because if err != nil is true, the function will return early.
```

Reply:
Resolved.